pkg/os/launchd/launchd_darwin.go

```diff
@@ -120,8 +125,8 @@ func AgentRunning(label string) bool {
 	if err != nil {
 		return false
 	}
-	if strings.TrimSpace(string(out)) == "-" {
-		return false
+	if match, err := regexp.MatchString(`^\d*$`, strings.TrimSpace(string(out))); err == nil && match {
```

Comment:
I'd add a small comment such as "match pid". 
`^\d*$` will match an empty string, I don't think we want that if we're looking for a pid.

Reply:
Ahh, thanks so much, should be `^\d+$`